packages/eth-json-rpc-middleware/CHANGELOG.md

```diff
@@ -9,6 +9,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- **BREAKING:** Migrate to `JsonRpcEngineV2` ([#7065](https://github.com/MetaMask/core/pull/7065))
+  - Migrates all middleware from `JsonRpcEngine` to `JsonRpcEngineV2`.
+  - Signatures of various middleware dependencies, e.g. `processTransaction` of `createWalletMiddleware`, have changed
```

Comment:
It might be worth it to detail this further to make upgrading easier. I'll do another pass when making a new release.